chore(roadmap): Enable development to be comparable via diff in a stable way

Dev2150 opened the pull request:
Problem: When edits are made (nodes are removed), reindexing happens, breaking the comparison
Solution: Add automatically stable IDs for every node

Dev2150 committed:
chore(tooling): Add graph comparison python script

Introduces a Python script, compare_develoment_roadmaps.py, designed to analyze and report the differences between two versions of .graphml files (currently designed for the develoment roadmap)

The script helps developers quickly identify:
- Added, removed, and common nodes (edges)
- Changes in node attributes
- Structural differences in the graph

When it comes to revisions it can be used:
- without any arguments (none specified)
- with one revision, e.g. HEAD
- with 2 revisions, e.g.
  - HEAD~1 HEAD~3
  - a1b2c3d 9f8e7d6

This tool is essential for tracking the evolution of graphs and for debugging issues that arise from graph modifications and for verifying changes before they are merged.

The tool is used by triggering a command such as 'git diff -- "MEIOUandTaxes/Documentation/Development roadmap.graphml"'

In order to make use of the tool even easier by using alias like:
- diff-roadmap
- diff-roadmap HEAD~3
- diff-roadmap HEAD~1 HEAD~3
- diff-roadmap a1b2c3d 9f8e7d6

the following lines are to be appended to the file .git/config:
"[alias]
 	diff-roadmap = "!sh -c 'git diff \"$@\" -- \"MEIOUandTaxes/Documentation/Development roadmap.graphml\"' :"
"

Dev2150 committed:
refactor: Split .graphml tooling files & moved to tools/ (for .git-blame-ignore-revs)

Dev2150 committed:
test: Make graphml comparer debuggable via environment variable

Dev2150 committed:
feat: Compare parents & show movement. Make comparisons more stable. Split code

Previous revision's comparison was unstable because upon .graphml edit, reindexing happens. The fix is to assign ID custom parameter to all nodes.

Dev2150 committed:
refactor(tooling): Move graphml scripts to separate repository

Dev2150 committed:
docs(roadmap): Add custom properties, e.g. ID, status, finished_date, priority, category, effort

Dev2150 commented:
Also moved the diff logic to another [repo](https://github.com/Dev2150/graphml-yed-diff-tool), cleaning the documentation folder.
New diff usage: https://github.com/MEIOU-and-Taxes/MnT-EU5/wiki#compare
If it is set up correctly, then the following changes can be tracked across commits:
- Connection (Edge)
- Name
- Parent (Group)
- Status
- Priority
- Category
- Effort
- Finished_date
- Target_release
- Assignee

MichaHelder committed:
Make mod appear in mod launcher

-For testing purposes, add metadata.json and the old thumbnail

MichaHelder committed:
Remove intermediary MEIOUandTaxes folder

- Mod can't be inside 2 folder, so just red rid of the extra one

MichaHelder committed:
Merge pull request #5 from MEIOU-and-Taxes/mepper

Make mod show up in-game

MichaHelder committed:
Add empty readme.md

MichaHelder committed:
Add a little text to readme to test GitHub tracker bot

Commit:
Removed pop promotion speed modifier from control.

Commit:
Remembered the common folder existed.

MichaHelder committed:
Use replace mode to prevent database conflict error

MichaHelder committed:
Merge branch 'malorn' into mepper

MichaHelder committed:
Don't let new born pops all be primary culture in cores

Just makes no sense

VineFynn committed:
Add files via upload

VineFynn committed:
Reduce speed of promote culture cabinet action

VineFynn committed:
Increase fishing village efficiency and decrease their total production to raise profitability

VineFynn committed:
Reduce culture promotion cabinet action speed to a couple of people a month

VineFynn committed:
Increase to 4 people per month

VineFynn committed:
provide small assimilation bonus in locations with culture buildings- we don't want it to be too large or there's an excessive incentive to urbanise areas you want to assimilate (although realistic, not desirable)

VineFynn committed:
Merge branch 'promote_culture_speed' of https://github.com/VineFynn/MnT-EU5 into promote_culture_speed

VineFynn committed:
Delete building_types directory

VineFynn committed:
Delete in_game/common/building_types/rural_buildings.txt

Commit:
Added production method to lumbermills which requires no inputs, just in case the whole economy shuts down.

VineFynn committed:
Makes fishing boats smaller, slightly more numerous, makes slightly more locations able to spawn them, and gives them a very small attack value

VineFynn committed:
Included impressed cogs as levies for the early game- I imagine we will want to be including later trade ships as well

VineFynn committed:
Not sure why this is here

VineFynn committed:
Shouldnt be there

VineFynn committed:
Remove content from different branch

VineFynn committed:
Merge pull request #2 from VineFynn/fishing_boats

Fishing boats

VineFynn committed:
Reference existing cog loc instead of copying it

VineFynn committed:
Use templating for cog levy

VineFynn committed:
Alter file names and refactor code to accommodate new code injection methods for EUV

VineFynn committed:
Make certain cultural buildings increase pop assimilation, reduce base assimilation by factor of ten

Commit:
Initialisation of vanilla files

state from version 1.0.3 to show changes

Commit:
Make River mapmode available in game

it was accessible from keybind, now it is available from regular mapmode menu

Commit:
Add two new mapmodes for potential tax base

They take current tax base and divide it by control. For provinces with exactly zero control, there is zero taxbase so calculation still returns zero. Others work well.

Commit:
Limited modifications instead of replacing the whole file

As suggested in pull review

VineFynn committed:
Use injection method rather than replacement

VineFynn committed:
change file name to reflect injection

VineFynn committed:
Replace REPLACE command with INJECT for ship building advance

Dev2150 committed:
chore(logs): Add logs for total population & list building types, for every region

MateuszGie committed:
Merge pull request #19 from MEIOU-and-Taxes/mapmodes_for_potential_taxbase

Mapmodes for potential taxbase and enabled river mapmode

Malorn-Deslor committed:
Merge pull request #15 from MEIOU-and-Taxes/malorn

Malorn balance changes to fix economy, specifically wood source if tools are missing, removing control penalty to pop promotion.

Malorn-Deslor committed:
Merge pull request #14 from VineFynn/develop

Changes to boat levies

MateuszGie committed:
Merge pull request #22 from MEIOU-and-Taxes/mepper

new born pops in cores will not be all of primary culture and removed pop promotion speed modifier from control, both using REPLACE mode

molkemon committed:
fix pops not giving building levels in any reasonable way

Commit:
Bailiffs now allowed in towns and cities.

Commit:
Improved sailors from wharfs, given how limited they are.

Commit:
Proximity rework, reduces 25% less rapidly, but provides less control.

Commit:
Added replace to wharf change.

molkemon committed:
add an auto-modifier that adds double your pop promotion speed to the (mostly unused) pop demotion speed modifier to fix extremely developed countries getting turbofucked in the last age by lack of laborers

CorlisFataneya commented:
Whats the Situation here now?

Dev2150 committed:
chore(logs): Log monthly (w/ regional prices) & plot & filter in PyQt6, with hotkeys

Dev2150 commented:
@CorlisFataneya 
> Whats the Situation here now?

waiting for approval

Dev2150 committed:
chore(metadata): Add game_id in metadata, so that Jetbrains plugin 'Paradox Language Support' correctly associates the mod with EU5

Malorn-Deslor committed:
Merge pull request #24 from molkemon/stopgap_measures

Stopgap Measures for two (imo) severe vanilla problems

Commit:
Merge remote-tracking branch 'origin/develop' into malorn

Malorn-Deslor committed:
Merge pull request #25 from MEIOU-and-Taxes/malorn

Proximity spread out more, providing less control.

Jarkami commented:
This does need someone to review, but I have simply not had the energy to take a proper look at it over the past couple of weeks. It's certainly free for someone else to take a look at instead

Dev2150 committed:
fix(logs): Log yearly + fix tooltip positioning + fix removing first moment having 0 buildings

VineFynn committed:
Vassals of the top overlord's primary culture cannot use the assimilate culture cabinet action

Overlords can use the assimilation cabinet action in their subject's lands

Rural settlements have much slower assimilation

This may be an overcorrection, but we'll see

VineFynn committed:
Devastation reduces assimilation speed

VineFynn committed:
Double war exhaustion from battles, to get feedback from others

Dev2150 committed:
Merge pull request #26 from Dev2150/Compatibility-with-Paradox-Support-Language

Add game_id in metadata, for Jetbrains plugin Paradox Language Support

Veheled reviewed (approved):
Looks fine to me!

VineFynn committed:
Merge pull request #35 from VineFynn/assimilation_speed

Changes to slow down assimilation speed, add some assimilation from cultural buildings.

VineFynn committed:
Merge pull request #36 from VineFynn/war_exhaustion

Triple war exhaustion from battles- feedback wanted

Dev2150 committed:
Merge pull request #21 from Dev2150/Log

Add logs for total population, building type count & prices, for every region + Add PyQt6 plotter with filters & hotkeys

Dev2150 committed:
Merge branch 'develop' of https://github.com/Dev2150/MnT-EU5 into Dev2150-develop

Dev2150 committed:
Merge branch 'Dev2150-develop' into develop
